src/managers/MerkleProofManager.sol

```diff
@@ -115,15 +116,18 @@ contract MerkleProofManager is IMerkleProofManager, IUpdateContract {
 }
 
 contract MerkleProofManagerFactory is IMerkleProofManagerFactory {
-    address public immutable spoke;
+    ISpoke public immutable spoke;
 
-    constructor(address spoke_) {
+    constructor(ISpoke spoke_) {
         spoke = spoke_;
     }
 
     /// @inheritdoc IMerkleProofManagerFactory
     function newManager(PoolId poolId) external returns (IMerkleProofManager) {
-        MerkleProofManager manager = new MerkleProofManager{salt: bytes32(uint256(poolId.raw()))}(poolId, spoke);
+        require(spoke.isPoolActive(poolId), InvalidPoolId());
```

Comment:
Curious, why the check? Could not ideally create the manager before having been notified by the hub that the pool is active?

Reply:
It came out of https://github.com/electisec-fellowship-block-7/centrifuge-v3-vaults-report/issues/5

I think it can be confusing if anyone can just create a manager for any ID. It might also be problematic for the indexer, as it would need to check whether a manager actually needs to be tracked